generate_directory_indexes.py

```diff
@@ -115,47 +156,131 @@ def format_directory_listing(directory_listing):
     })
     return out
 
-def walk(path, base_path):
-    contents = os.listdir(path)
-    directory_listing = []
+def file_information(full_path, file, last_modified = None, size = None):
+    if last_modified == None:
+        last_modified = os.path.getmtime(full_path)
+    if size == None:
+        size =  os.path.getsize(full_path)
+    icon = 'folder.gif' if os.path.isdir(full_path) else 'unknown.gif'
+
+    return {
+        'name': file,
+        'lastModified': last_modified,
+        'size': size,
+        'icon': icon
+    }
+
+def string_to_epoch_seconds(string, format):
+    utc_time = datetime.strptime(string, format)
+    epoch_time = (utc_time - datetime(1970, 1, 1)).total_seconds()
+
+    return epoch_time
+
+
+def is_an_index_file(file_name):
```

Comment:
Does python support question marks in function defs?

Reply:
It doesn't.